app/services/IncomeSourceDetailsService.scala

```diff
@@ -50,23 +51,44 @@ class IncomeSourceDetailsService @Inject()(val incomeTaxViewChangeConnector: Inc
 
   def getIncomeSourceDetails(cacheKey: Option[String] = None)(implicit hc: HeaderCarrier,
                                                               mtdUser: MtdItUserWithNino[_]): Future[IncomeSourceDetailsResponse] = {
-    if (cacheKey.isDefined) {
-      getCachedIncomeSources(cacheKey.get).flatMap {
-        case Some(sources: IncomeSourceDetailsModel) =>
-          Logger("application").info(s"incomeSourceDetails cache HIT with ${cacheKey.get}")
-          Future.successful(sources)
-        case None =>
-          Logger("application").info(s"incomeSourceDetails cache MISS with ${cacheKey.get}")
-          incomeTaxViewChangeConnector.getIncomeSources().flatMap {
-            case incomeSourceDetailsModel: IncomeSourceDetailsModel =>
-              cache.set(cacheKey.get, incomeSourceDetailsModel.sanitise.toJson, cacheExpiry).map(
-                _ => incomeSourceDetailsModel
-              )
-            case error: IncomeSourceDetailsResponse => Future.successful(error)
-          }
-      }
-    } else {
+//    if (cacheKey.isDefined) {
+//      getCachedIncomeSources(cacheKey.get).flatMap {
+//        case Some(sources: IncomeSourceDetailsModel) =>
+//          Logger("application").info(s"incomeSourceDetails cache HIT with ${cacheKey.get}")
+//          Future.successful(sources)
+//        case None =>
+//          Logger("application").info(s"incomeSourceDetails cache MISS with ${cacheKey.get}")
+//          incomeTaxViewChangeConnector.getIncomeSources().flatMap {
+//            case incomeSourceDetailsModel: IncomeSourceDetailsModel =>
+//              cache.set(cacheKey.get, incomeSourceDetailsModel.sanitise.toJson, cacheExpiry).map(
+//                _ => incomeSourceDetailsModel
+//              )
+//            case error: IncomeSourceDetailsResponse => Future.successful(error)
+//          }
+//      }
+//    } else {
       incomeTaxViewChangeConnector.getIncomeSources()
-    }
+//    }
+  }
+
+  def incomeSourcesAsViewModel(sources: IncomeSourceDetailsModel): IncomeSourcesViewModel = {
+    IncomeSourcesViewModel(
+      soleTraderBusinesses = sources.businesses.filterNot(_.isCeased).map {
+        case b if b.tradingName.nonEmpty && b.tradingStartDate.nonEmpty =>
```

Comment:
instead of using variable name p / b - use full names, property and business name accordingly, this would simplify reading/maintaining this code in the future